profiling/src/string_table/borrowed.rs

```diff
@@ -0,0 +1,80 @@
+use ahash::RandomState;
+use std::collections::HashMap;
+use std::ops::Range;
+
+pub struct BorrowedStringTable<'a> {
+    pub vec: Vec<&'a str>,
+    pub map: HashMap<&'a str, usize, RandomState>,
+}
+
+impl<'a> Default for BorrowedStringTable<'a> {
+    fn default() -> Self {
+        /// The initial size of the Vec. At the time of writing, Vec would
+        /// choose size 4. This is expected to be much too small for the
+        /// use-case, so use a larger initial capacity to save a few
+        /// re-allocations in the beginning.
+        /// This is just an educated estimate, not a finely tuned value.
+        const INITIAL_VEC_CAPACITY: usize = 1024 / std::mem::size_of::<&str>();
```

Comment:
@morrisonlevi just for my understanding: assuming we compile for a 64 bit platform, `size_of::<&str>()` is 8 (64 bit), so we reserve initial memory for 128 string references, on a 32 bit platform we would initially reserve memory for 256 string references. Did you mean to pre alloc space for 1024 string references?

Reply:
`sizeof::<&str>()` is actually 16 bytes; this is a "fat pointer." The capacity is therefore only 64 strings, and on 32-bit (which we don't support) it would be double that. The general goals were:

- Keep it within 1 page of memory.
- Reduce allocations during the startup, which already has a lot of memory churn.

Honestly, this `1024` number should probably be `2048` -- half a page of memory. Then the `HashMap` would use roughly the other half of the page (logically, not guaranteed by any allocator).